### ISSUE TITLE
User-configurable stack name in init-odkx-sync-endpoint.py

### ISSUE THREAD

**Issue by linl33** [odk-x/tool-suite-X#194]
`init-odkx-sync-endpoint.py` hard-coded the stack name `syncldap`. A user-configurable stack name would be useful.

**Comment by adityaofficial10** [comment]
Hey!
@linl33 Can I work on this issue?

**Comment by linl33** [comment]
Hi @adityaofficial10!

Yes, you can work on this issue.
The idea is that the script should prompt for a stack name but default to `syncldap` (the current name). The script should also store the chosen stack name to its settings file.

**Comment by adityaofficial10** [comment]
@linl33 
Sure, I'll follow this up with a PR shortly.

**Comment by adityaofficial10** [comment]
@linl33 
I have made the changes as shown below:
<img width="862" alt="Screenshot 2021-03-05 at 11 59 35 PM" src="https://user-images.githubusercontent.com/65034276/110158079-f2052f00-7e0e-11eb-9110-37e465453774.png">
Is this fine?
Also, you were talking about the settings file.
Which file is that exactly?

**Comment by linl33** [comment]
The function is correct but a few notes on the implementation,
 - Please move the prompt into `run_interactive_config`, to keep it consistent with the rest of the configuration
 - Please put the default value into `[()]`, e.g. `Please enter a name for the Docker Swarm stack [(syncldap)]`
 - Please use String interpolation instead of concatenation

For the settings file, see https://github.com/odk-x/sync-endpoint-default-setup/blob/0d97f322b6f3c48f86cdb47c9c1c7acf3f7bfb93/init-odkx-sync-endpoint.py#L134

Let me know if you have questions. Thanks for the work so far!

**Comment by adityaofficial10** [comment]
@linl33 
I've made the required changes.
I'm sending you a link to the file in my fork. Kindly review and inform if I can improve something.
https://github.com/adityaofficial10/sync-endpoint-default-setup/blob/fix/stack-name-added/init-odkx-sync-endpoint.py

**Comment by linl33** [comment]
Yes, please send a pull request against the **development** branch.

**Comment by adityaofficial10** [comment]
@linl33 
I've linked the PR.

**Comment by wbrunette** [comment]
For future reference for someone taking on this issue. Take a look at the previous incomplete PR. https://github.com/odk-x/sync-endpoint-default-setup/pull/26

The comments in the PR give additional direction

**Comment by wsmmxmm** [comment]
Hi team! I am an outreachy applicant. Please can I work on this issue if it is still open?

**Comment by wsmmxmm** [comment]
Hi team! I am an outreachy applicant. Please can I work on this issue if it is still open?

**Comment by wbrunette** [comment]
@wsmmxmm it is still open. I will assign it to you. Check out the PR I commented about as much of the work has been attempted once, just not fully completed.

**Comment by Ndacyayisenga-droid** [comment]
@wbrunette @linl33 @elmps2018
Can I finish up with this if its still open?
Regards,
Noah

**Comment by MariaAngella** [comment]
Hello team,

Can I take this on please, I'm an Outreachy applicant.

Thank you and regards,
Maria Angella

**Comment by elmps2018** [comment]
Hi @MariaAngella! @wsmmxmm is currently assigned but does not look like they worked on it lately. @wbrunette @wsmmxmm is it okay if I reassign?

**Comment by wbrunette** [comment]
You should first ask the question if they want to have a custom stack name. Because if they change it the directions will not match. So really it's two questions.

**Comment by MariaAngella** [comment]
Thank you for this assignment.
